src/configupdater/option.py

```diff
@@ -95,10 +96,26 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f"<Option: {self._key} = {self.value!r}>"
 
+    def _instantiate_copy(self: T) -> T:
+        """Will be called by :meth:`Block.__deepcopy__`"""
+        self._join_multiline_value()
+        return self.__class__(
+            self._key,
+            self._value,
+            container=None,
+            delimiter=self._delimiter,
+            space_around_delimiters=self._space_around_delimiters,
+        )
+
+    def optionxform(self, optionstr: str) -> str:
+        if self.has_container():
+            section = cast("Section", self.container)
+            return section.optionxform(optionstr)
+        return optionstr.lower()
+
```

Comment:
The problem with having optionxform not on the document level is that a user would need to overwrite this method in `Block`, `Section`, etc. See the docs of ConfigParser why they choose this hook for the user.